src/constants/mediaQueries/index.ts

```diff
@@ -9,4 +9,8 @@ const mediaQueries = Object.entries(BREAKPOINTS).reduce(
   {}
```

Comment:
It looks like [Augmenting key name during mapping](https://github.com/microsoft/TypeScript/issues/12754) isn't supported by TypeScript but we can cheat by casting this {} like so:

```
const mediaQueries = Object.entries(BREAKPOINTS).reduce(
  (accumulator, [label, px]) => ({
    ...accumulator,
    [`${label}Up`]: `@media (min-width: ${px}px)`,
  }),
  {} as {
    xsUp: string;
    smUp: string;
    mdUp: string;
    lgUp: string;
    xlUp: string;
    xxlUp: string;
  },
);
```

The return value of `throwOnUndefinedProperty(mediaQueries)` will have only those specific keys as options, instead of any string key, which will allow us to catch errors at compile-time instead of relying on the `throwOnUndefinedProperty` runtime error.

Reply:
Ok so thanks for bringing this up. I remember not being happy about not having intellisense and strong type for mediaqueries.
Due to the rare occasion that we need to add a new breakpoint I just decided to hardcode the Type so that we can have strong type and intellisense.

I left a comment  in breakpoint file that if a new one is added the mediaquery type needs to be udpated